MAST: remote test failure with roman

@snbianco - could you please have a look into this? It looks like hitting an astropy issue, so either a problem with the test as it only shows for roman, but also maybe something to be fixed in the code.



```
Numpy: 2.2.6
Matplotlib: 3.10.3
Astropy: 7.1.0
regions: 0.10
pyVO: 1.7
mocpy: 0.18.0
astropy-healpix: 1.1.2
vamdclib: not available
```


```
____________________________________________________________ TestMast.test_missions_workflow[roman-query_params3] ____________________________________________________________

self = <astroquery.mast.tests.test_mast_remote.TestMast object at 0x13eb80b60>
tmp_path = PosixPath('/private/var/folders/9s/070g0pd502q70k3gffpxv8km0000gq/T/pytest-of-bsipocz/pytest-551/test_missions_workflow_roman_q0'), mission = 'roman'
query_params = {'program': 3}

    @pytest.mark.parametrize("mission, query_params", [
        ('jwst', {'fileSetName': 'jw01189001001_02101_00001'}),
        ('classy', {'Target': 'J0021+0052'}),
        ('ullyses', {'host_galaxy_name': 'WLM', 'select_cols': ['observation_id']}),
        ('roman', {'program': 3}),
    ])
    def test_missions_workflow(self, tmp_path, mission, query_params):
        # Test workflow with other missions
        m = MastMissions(mission=mission)
    
        # Roman requires extra setup to point towards the test server
        if mission == 'roman':
            m._service_api_connection.SERVICE_URL = 'https://masttest.stsci.edu'
            m._service_api_connection.REQUEST_URL = 'https://masttest.stsci.edu/search/roman/api/v0.1/'
    
        # Criteria query
>       datasets = m.query_criteria(**query_params)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

astroquery/mast/tests/test_mast_remote.py:376: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astroquery/utils/class_or_instance.py:25: in f
    return self.fn(obj, *args, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
astroquery/utils/process_asyncs.py:35: in newmethod
    result = self._parse_result(response, verbose=verbose)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
astroquery/mast/missions.py:134: in _parse_result
    results = self._service_api_connection._parse_result(response, verbose, data_key='results')
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
astroquery/mast/services.py:236: in _parse_result
    result_table = _json_to_table(result, data_key=data_key)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
astroquery/mast/services.py:105: in _json_to_table
    data_table.add_column(MaskedColumn(col_data.astype(col_type), name=col_name, mask=col_mask))
../../../.pyenv/versions/3.12.1/lib/python3.12/site-packages/astropy/table/table.py:2461: in add_column
    self.columns[name] = col
    ^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <TableColumns names=('pass','ArchiveFileID','fileSetName','detector','productLevel','ra_ref','dec_ref','product_type','program_category','program_subcategory','program_title','exposure_type')>
item = 'program_subcategory'
value = <MaskedColumn name='program_subcategory' dtype='str4' length=18>
None
None
None
None
None
None
None
None
None
None
None
None
None
None
None
None
None
None
validated = False

    def __setitem__(self, item, value, validated=False):
        """
        Set item in this dict instance, but do not allow directly replacing an
        existing column unless it is already validated (and thus is certain to
        not corrupt the table).
    
        NOTE: it is easily possible to corrupt a table by directly *adding* a new
        key to the TableColumns attribute of a Table, e.g.
        ``t.columns['jane'] = 'doe'``.
    
        """
        if item in self and not validated:
>           raise ValueError(
                f"Cannot replace column '{item}'.  Use Table.replace_column() instead."
            )
E           ValueError: Cannot replace column 'program_subcategory'.  Use Table.replace_column() instead.

../../../.pyenv/versions/3.12.1/lib/python3.12/site-packages/astropy/table/table.py:292: ValueError
========================================================================== short test summary info ===========================================================================
FAILED astroquery/mast/tests/test_mast_remote.py::TestMast::test_missions_workflow[roman-query_params3] - ValueError: Cannot replace column 'program_subcategory'.  Use Table.replace_column() instead.

```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -130,6 +130,8 @@ mast
 
 - Enhanced ``filter_products`` methods in ``MastMissions`` and ``Observations`` to support filtering with negated values. [#3393]
 
+- Fix bug where duplicate columns from server responses cause an error when converting to an `~astropy.table.Table`. [#3400]
+
 
 Infrastructure, Utility and Other Changes and Additions
 -------------------------------------------------------

--- a/astroquery/mast/discovery_portal.py
+++ b/astroquery/mast/discovery_portal.py
@@ -101,8 +101,9 @@ def _json_to_table(json_obj, col_config=None):
         else:
             col_mask = np.equal(col_data, ignore_value)
 
-        # add the column
-        data_table.add_column(MaskedColumn(col_data.astype(atype), name=col, mask=col_mask))
+        # add the column if it does not exist already
+        if col not in data_table.colnames:
+            data_table.add_column(MaskedColumn(col_data.astype(atype), name=col, mask=col_mask))
 
     return data_table
 

--- a/astroquery/mast/services.py
+++ b/astroquery/mast/services.py
@@ -101,8 +101,9 @@ def _json_to_table(json_obj, data_key='data'):
         else:
             col_mask = np.equal(col_data, ignore_value)
 
-        # add the column
-        data_table.add_column(MaskedColumn(col_data.astype(col_type), name=col_name, mask=col_mask))
+        # add the column if it does not exist already
+        if col_name not in data_table.colnames:
+            data_table.add_column(MaskedColumn(col_data.astype(col_type), name=col_name, mask=col_mask))
 
     return data_table
 

